Simulation/Classes/ControlsVector_Index.m

```diff
@@ -0,0 +1,11 @@
+%% Controls Vector Indices
+% This is an enumeration of all Indices of a Vehicle Controls Vector
+
+% Note: enumerations are currently not implemented in Octave
+% (https://savannah.gnu.org/bugs/?44582) so constant properties are used instead.
+
+classdef ControlsVector_Index

```

Comment:
My original idea was to implement the the controls vector on the fly so models can create controls on the fly. 

But this made me realize if two models were both to use the `1` position in the vector for controls, then you would have to adjust the controls inputs for one of the models, which shouldn't be the case...

However, doing it this way also can turn into a bad thing since every new control model may have their own variation of controls so it could just be adding and adding new indices to the list. The StateVector is somewhat immune to this due to most vehicles having the majority of states shared such as position, velocity, mass, and orientation.

What do you think of instead of calling it a "Controls Vector" we call it a "Controls Structure"? Then we have every model take in a "Controls Structure" and the controls for that model is then pulled from the fields in the structure such as:

```
controlsStruct = struct;
controlsStruct.Throttle = 0;

...

dervs = ElectripPropellerModel(stateVector(controlsStruct);
```
What do you think? I'm okay with either method (keeping as is, or switching to a structure) but wanted to throw out that option as well.

Reply:
Resolving this for now as the solution we came up with is outside of the scope of this PR.